fixes #520 (#568)
cds-hooks/docs: docs/specification/current.md

#### Prefetch tokens

A prefetch token is a placeholder in a prefetch template that is *_replaced by information from the hook's context_* to construct the FHIR URL used to request the prefetch data.

> Prefetch tokens MUST be delimited by `{{` and `}}`, and MUST contain only the qualified path to a hook context field *_or one of the following user identifiers: `userPractitionerId`, 'userPractitioneRolerId', `userPatientId`, or `userRelatedPersonId`_*.

Reviewer: minor typo
```suggestion
Prefetch tokens MUST be delimited by `{{` and `}}`, and MUST contain only the qualified path to a hook context field *_or one of the following user identifiers: `userPractitionerId`, 'userPractitioneRoleId', `userPatientId`, or `userRelatedPersonId`_*.
```

Individual hooks specify which of their `context` fields can be used as prefetch tokens. Only root-level fields with a primitive value within the `context` object are eligible to be used as prefetch tokens. For example, `{{context.medication.id}}` is not a valid prefetch token because it attempts to access the `id` field of the `medication` field.

##### Prefetch tokens identifying the user
A prefetch template enables a CDS Service to learn more about the current user through a FHIR read, like so:
```
{
  "prefetch": {
    "user": "{{context.userId}}"
  }
}
```
or though a FHIR search:
```
{
  "prefetch": {

>     "user": "PractitionerRole?_id={{userPractitionerId}}&_include=PractitionerRole:practitioner"

Reviewer: Wouldn't this be userPractitionerRoleId?
```suggestion
    "user": "PractitionerRole?_id={{userPractitionerRoleId}}&_include=PractitionerRole:practitioner"
```

  }
}
```

A prefetch template may include any of the following prefetch tokens:

Token | Description
---|---
`{{userPractitionerId}}` | FHIR id of the Practitioner resource corresponding to the current user. 

> `{{userPractitionerRoleId}}`|FHIR id of the Practitioner resource corresponding to the current user. 
> `{{userPatientId}}`|FHIR id of the Practitioner resource corresponding to the current user. 
> `{{userRelatedPersonId}}`|FHIR id of the Practitioner resource corresponding to the current user. 

Reviewer: Update "Practitioner" to be the appropriate type
```suggestion
`{{userPractitionerRoleId}}`|FHIR id of the PractitionerRole resource corresponding to the current user. 
`{{userPatientId}}`|FHIR id of the Patient resource corresponding to the current user. 
`{{userRelatedPersonId}}`|FHIR id of the RelatedPerson resource corresponding to the current user. 
```

Author: thank you, Dennis!

> No single FHIR resource represents a user, rather Practitioner and PractitionerRole may be jointly used to represent a provider, and Patient or Person are used to represent a patient or their proxy. Hook definitions typically define a `context.userId` field and corresponding prefetch token.

Author: ```suggestion
No single FHIR resource represents a user, rather Practitioner and PractitionerRole may be jointly used to represent a provider, and Patient or RelatedPerson are used to represent a patient or their proxy. Hook definitions typically define a `context.userId` field and corresponding prefetch token.
```

> No single FHIR resource represents a user, rather Practitioner and PractitionerRole may be jointly used to represent a provider or other, and Patient or Person are used to represent a patient or their proxy. Hook definitions typically define a `context.userId` field and corresponding prefetch token.

Reviewer: Did you mean RelatedPerson?
```suggestion
No single FHIR resource represents a user, rather Practitioner and PractitionerRole may be jointly used to represent a provider or other, and Patient or RelatedPerson are used to represent a patient or their proxy. Hook definitions typically define a `context.userId` field and corresponding prefetch token.
```

Reviewer: What does "or other" mean?  Should this be removed or am I misunderstanding?
```suggestion
No single FHIR resource represents a user, rather Practitioner and PractitionerRole may be jointly used to represent a provider, and Patient or Person are used to represent a patient or their proxy. Hook definitions typically define a `context.userId` field and corresponding prefetch token.
```
